06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * An inverted-index class
+ * @class
+ */
+class InvertedIndex {
+  /**
+  * class constructor
+  * @constructor
+  */
+  constructor() {
+    this.allIndex = {};
+    this.allFiles = {};
+    this.allLength = {};
+    this.allFilesTitle = [];
+    this.validatedFile = {};
+  }
+
+  /**
+   * Validate Method
+   * @param {Object} validateFile an object to be validated
+   * @returns {Object} file that has been validated
+   */
+  validateFileFunc(validateFile) {
+    this.validatedFile = validateFile;
+    if (validateFile.title !== undefined && validateFile.text !== undefined) {
+      return validateFile;
+    } return false;
+  }
+
+  /**
+   * Sanitizer
+   * @param {Object} BookObject recieves json object
+   * @return {Object} returns an Object of names
+   */
+  sanitizer(BookObject) {
+    Object.keys(BookObject).forEach((titles) => {
+      this.allFilesTitle.push(BookObject[titles].title);
+    });
+  }
+
+  /**
+   * Get all Indecies
+   * @param {string} bookName name of the individaul book
+   * @returns {Object} allIndicies returns all indexed file
+   */
+  getAllIndecies(bookName) {
```

Comment:
`getAllIndecies` sounds like you're getting all the created indices.

I'm not so sure I see the value in returning false when the parameter is undefined, as agains checking if the index with that parameter actually exists.

Also, is false the best thing to return? #ThinkAboutIt